src/Symfony/Component/Config/Loader/FileLoader.php

```diff
@@ -83,10 +83,7 @@ public function import($resource, $type = null, $ignoreErrors = false, $sourceRe
             $loader = $this->resolve($resource, $type);
 
             if ($loader instanceof self && null !== $this->currentDir) {
-                // we fallback to the current locator to keep BC
-                // as some some loaders do not call the parent __construct()
-                // @deprecated should be removed in 3.0
-                $locator = $loader->getLocator() ?: $this->locator;
+                $locator = $loader->getLocator();
                 $resource = $locator->locate($resource, $this->currentDir, false);
```

Comment:
well, not calling the parent constructor to initialize the locator in classes extending FileLoader is not supported anymore in 3.0. the object should be built in a working state.

Btw, it looks like we are missing a deprecation warning here to warn about that in 2.x versions. This should be added *before* removing deprecated stuff from 3.0